crates/bevy_render/macros/src/as_bind_group.rs

```diff
@@ -0,0 +1,389 @@
+                value @ BindingState::Free => {
+                    *value = match binding_type {
+                        BindingType::Uniform => BindingState::OccupiedMergableUniform {
+                            uniform_fields: vec![field],
+                        },
+                        _ => {
+                            // only populate bind group entries for non-uniforms
+                            // uniform entries are deferred until the end
+                            let binding_vec_index = bind_group_entries.len();
+                            bind_group_entries.push(quote! {
+                                #render_path::render_resource::BindGroupEntry {
+                                    binding: #binding_index,
+                                    resource: bindings[#binding_vec_index].get_binding(),
+                                }
+                            });
+                            BindingState::Occupied {
+                            binding_type,
+                            ident: field_name,
+                        }},
+                    }
+                },
+                BindingState::Occupied { binding_type, ident: occupied_ident} => panic!(
+                    "The '{field_name}' field cannot be assigned to binding {binding_index} because it is already occupied by the field '{occupied_ident}' of type {binding_type:?}."
+                ),
+                BindingState::OccupiedConvertedUniform => panic!(
+                    "The '{field_name}' field cannot be assigned to binding {binding_index} because it is already occupied by a struct-level uniform binding at the same index."
+                ),
+                BindingState::OccupiedMergableUniform { uniform_fields } => {
+                    match binding_type {
+                        BindingType::Uniform => {
+                            uniform_fields.push(field);
+                        },
+                        _ => {panic!("The '{field_name}' field cannot be assigned to binding {binding_index} because it is already occupied by a {:?}.", BindingType::Uniform)},
+                    }
+                },
+            }
+
+            match binding_type {
+                BindingType::Uniform => { /* uniform codegen is deferred to account for combined uniform bindings */
+                }
+                BindingType::Texture => {
+                    binding_impls.push(quote! {
+                        #render_path::render_resource::OwnedBindingResource::TextureView({
+                            let handle: Option<&#asset_path::Handle<#render_path::texture::Image>> = (&self.#field_name).into();
+                            if let Some(handle) = handle {
+                                images.get(handle).ok_or_else(|| #render_path::render_resource::AsBindGroupError::RetryNextUpdate)?.texture_view.clone()
+                            } else {
+                                fallback_image.texture_view.clone()
```

Comment:
At some point I expect that we will need to add a possibility to define the value of the fallback image per-texture as I think they may not always need to be white but rather some other value.

Reply:
agreed!

Reply:
(resolving for now though because its not "required functionality")